cmake_test/add_section.cmake

```diff
@@ -0,0 +1,41 @@
+include_guard()
+
+
+#[[[
+# Adds a test section, should be called inside of a declared test function directly before declaring the section function.
+# The NAME parameter will be populated as by set() with the generated section function name. Declare the section function using this generated name. Ex:
+#
+# .. code-block:: cmake
+#
+#    #This is inside of a declared test function
+#    ct_add_section(NAME this_section)
+#    function(${this_section})
+#        message(STATUS "This code will run in a test section")
+#    endfunction()
+#
+# :param EXPECTFAIL: Option indicating whether the section is expected to fail or not, if specified will cause the section to be ran in a subprocess.
+# :param NAME name: Required argument specifying the name variable of the section. Will set a variable with specified name containing the generated function ID to use.
+#]]
+macro(ct_add_section)
+
+    #TODO Set sections as a subproperty of CT_CURRENT_EXECUTION_UNIT instead of as a single global variable
+    set(options EXPECTFAIL)
+    set(oneValueArgs NAME)
+    set(multiValueArgs "")
+    cmake_parse_arguments(CT_ADD_SECTION "${options}" "${oneValueArgs}"
```

Comment:
This probably needs to be documented [here](https://cmakedev.readthedocs.io/en/latest/conventions/code_conventions.html#) (I swear it was documented somewhere, but now I can't find it), but please use snake_case for your variables and prefix them with an underscore and the first letters of the function name (ignoring the namespace). So all variables in this macro should be prefixed with `_as`.

Reply:
This should now be fixed.